internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -1635,70 +1635,61 @@ func TestComputeMachineDeployment(t *testing.T) {
 			}).
 			Build()
 
-		machineDeploymentStable := builder.MachineDeployment("test-namespace", "md-1").
-			WithGeneration(1).
-			WithReplicas(2).
-			WithStatus(clusterv1.MachineDeploymentStatus{
-				ObservedGeneration: 2,
-				Replicas:           2,
-				UpdatedReplicas:    2,
-				AvailableReplicas:  2,
-			}).
-			Build()
-
-		machineDeploymentRollingOut := builder.MachineDeployment("test-namespace", "md-1").
-			WithGeneration(1).
-			WithReplicas(2).
-			WithStatus(clusterv1.MachineDeploymentStatus{
-				ObservedGeneration: 2,
-				Replicas:           1,
-				UpdatedReplicas:    1,
-				AvailableReplicas:  1,
-			}).
-			Build()
-
-		machineDeploymentsStateRollingOut := scope.MachineDeploymentsStateMap{
-			"class-1": &scope.MachineDeploymentState{Object: machineDeploymentStable},
-			"class-2": &scope.MachineDeploymentState{Object: machineDeploymentRollingOut},
-		}
-
 		// Note: in all the following tests we are setting it up so that the control plane is already
 		// stable at the topology version.
 		// A more extensive list of scenarios is tested in TestComputeMachineDeploymentVersion.
 		tests := []struct {
-			name                    string
-			machineDeploymentsState scope.MachineDeploymentsStateMap
-			currentMDVersion        *string
-			topologyVersion         string
-			expectedVersion         string
+			name                        string
+			upgradingMachineDeployments []string
+			currentMDVersion            *string
+			upgradeConcurrency          string
+			topologyVersion             string
+			expectedVersion             string
 		}{
 			{
-				name:                    "use cluster.spec.topology.version if creating a new machine deployment",
-				machineDeploymentsState: nil,
-				currentMDVersion:        nil,
-				topologyVersion:         "v1.2.3",
-				expectedVersion:         "v1.2.3",
+				name:                        "use cluster.spec.topology.version if creating a new machine deployment",
```

Comment:
Can we also have an additional case where we have an upgrading MD but the new MD is created anyway with the topologyVersion?